libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -922,7 +922,7 @@ string YulUtilFunctions::readFromCalldata(Type const& _type)
 	return readFromMemoryOrCalldata(_type, true);
 }
 
-string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::optional<unsigned> const _offset)
+string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::optional<unsigned> const& _offset)
```

Comment:
I'm guessing this change and the one below it does not belong to this PR. Could it be moved to #7077 ?

Reply:
Without this change, this PR would fail in the CI. Basically, this is the problem that was previously undetected that the CI will now test for

Reply:
so it does indeed belong together

Reply:
okay, looks good to me. may be good to edit PR title to something "Fix build error and add ... gcc-7"